### PR TITLE
test: Buildkite client tests

### LINKED ISSUE #766: Buildkite: Client tests using Mock Server
**Description**
Implement tests using for Buildkite client using Buildkite Mock server.

**Acceptance Criteria**
- [ ] Tests implemented to cover all Buildkite client methods using Buildkite Mock server.

**Additional context**
[Jenkins Client tests using Mock server](https://github.com/platform-platform/monorepo/blob/master/metrics/ci_integrations/test/client/jenkins/jenkins_client_test.dart)
[Testing Third Party API](https://github.com/platform-platform/monorepo/blob/master/docs/03_third_party_api_testing.md)

### PR DESCRIPTION
**Pull request by solid-maxim**
Fixes #766 

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: metrics/ci_integrations/test/client/buildkite/test_utils/mock/buildkite_mock_server.dart [hunk 1]
```diff
@@ -159,6 +165,18 @@ class BuildkiteMockServer extends ApiMockServer {
     );
   }
 
+  /// Returns the `finished_from` query parameter of the given [request].
+  ///
+  /// Returns `null` if the [finishedFrom] is `null` or the [DateTime] can't be
+  /// parsed from the given [finishedFrom].
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/test/client/buildkite/test_utils/mock/buildkite_mock_server.dart:171]:
```suggestion
  /// Returns `null` if the `finishedFrom` is `null` or the [DateTime] can't be
  /// parsed from the given `finishedFrom`.
```

File: metrics/ci_integrations/test/client/buildkite/test_utils/mock/buildkite_mock_server.dart [hunk 2]
```diff
@@ -128,10 +129,15 @@ class BuildkiteMockServer extends ApiMockServer {
     return const BuildkiteBuildStateMapper().map(state);
   }
 
-  /// Generates a list of [BuildkiteBuild]s with the given [state].
+  /// Generates a list of [BuildkiteBuild]s with the given [state]
+  /// and [finishedFrom].
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/test/client/buildkite/test_utils/mock/buildkite_mock_server.dart:133]: Should we update the documentation of the `_pipelineBuildsResponse` method to include info about the `finishedFrom` parameter?

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Great job, @solid-daniyilyevtyushkin & @solid-maxim! I've added a couple of suggestions here - please take a look.

File: metrics/ci_integrations/test/client/buildkite/buildkite_client_test.dart [hunk 3]
```diff
@@ -127,12 +147,406 @@ void main() {
...
+    );
+
+    test(
+      ".fetchBuildsNext() returns a next builds page after the given one",
+      () async {
+        final interactionResult = await client.fetchBuilds(
+          pipelineSlug,
+          page: testPageNumber,
+        );
+        final buildsPage = interactionResult.result;
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/test/client/buildkite/buildkite_client_test.dart:338]:
```suggestion
        const buildsPage = BuildkiteBuildsPage(
          page: 1,
          perPage: 1,
        );
```
Does it make sense?

File: metrics/ci_integrations/test/client/buildkite/buildkite_client_test.dart [hunk 4]
```diff
@@ -127,12 +147,406 @@ void main() {
...
+    );
+
+    test(
+      ".fetchArtifacts() fails if an associated build with such number is not found",
+      () async {
+        const buildNumber = 10;
+        final interactionResult = await client.fetchArtifacts(
+          pipelineSlug,
+          buildNumber,
+        );
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/test/client/buildkite/buildkite_client_test.dart:365]: Maybe we can inline `buildNumber` here?

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Almost there, @solid-daniyilyevtyushkin & @solid-maxim! I've added a few suggestions here - hope they make sense.

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @solid-maxim  & @solid-daniyilyevtyushkin!

**Comment by solid-daniyilyevtyushkin** [comment]
@Mergifyio refresh

**Pull request merged**